link.go

```diff
@@ -90,38 +89,15 @@ func (l *link) attach(ctx context.Context, beforeAttach func(*frames.PerformAtta
 	// send Attach frame
 	debug.Log(1, "TX (attachLink): %s", attach)
 
-	// we use send to have positive confirmation on transmission
-	send := make(chan encoding.DeliveryState)
-	_ = l.session.txFrame(attach, send)
+	_ = l.session.txFrame(attach, nil)
 
 	// wait for response
 	var fr frames.FrameBody
 	select {
 	case <-ctx.Done():
-		select {
-		case <-send:
-			// attach was written to the network. assume it was received
-			// and that the ctx was too short to wait for the ack. in this
-			// case we must send a detach before deallocation
-			go func() {
-				_ = l.session.txFrame(&frames.PerformDetach{
-					Handle: l.handle,
-					Closed: true,
-				}, nil)
-				select {
-				case <-l.session.done:
-					// session has terminated, no need to deallocate in this case
-				case <-time.After(5 * time.Second):
-					debug.Log(3, "link.attach() clean-up timed out waiting for ack")
-				case <-l.rx:
-					// received ack, safe to delete handle
-					l.session.deallocateHandle(l)
-				}
-			}()
-		default:
-			// attach wasn't written to the network, so delete the handle
-			l.session.deallocateHandle(l)
-		}
+		// attach was written to the network. assume it was received
+		// and that the ctx was too short to wait for the ack.
+		go l.muxDetach(nil, nil)
```

Comment:
Can this lead to a goroutine that never exits? It seems like we need some sort of timeout since it's possible we never actually sent the attach, correct?

Reply:
Yes, and it really highlights the problem we have with our current implementation.  I will add a timeout.